### ISSUE TITLE
60 timeout on Elastic Beanstalk using the old AL1

### ISSUE THREAD

**Issue by aguerard** [metabase/metabase#19383]
NOTE: https://github.com/metabase/metabase/pull/19387 I have a PR that fixed my issue. I believe it's specific to Elastic Beanstalk applications using the old, "AL1" (pre AL2) Docker Platform.

**Describe the bug**
Queries that previously didn't timeout are now timing out after upgrading to v0.41.4 from v0.38.1. 

One of our queries used to display in Metabase fine prior, but now appears to be timing out. Using RDS insights, it looks like the query is timing out around 60seconds. I'm able to successfully run the query (it takes 75seconds) from another client (e.g. MySQLWorkbench).

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)

```
[fb91212e-1108-4858-ad1e-cb85839ebec0] 2021-12-15T08:40:33-08:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 2,
 :started_at #t "2021-12-15T16:39:32.131894Z[GMT]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Error executing query",
   :stacktrace
   ["--> driver.sql_jdbc.execute$execute_reducible_query$fn__52437.invoke(execute.clj:504)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:501)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:487)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:496)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:487)"
    "driver.sql_jdbc$fn__85100.invokeStatic(sql_jdbc.clj:54)"
    "driver.sql_jdbc$fn__85100.invoke(sql_jdbc.clj:52)"
    "query_processor.context$executef.invokeStatic(context.clj:59)"
    "query_processor.context$executef.invoke(context.clj:48)"
    "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
    "query_processor.context.default$default_runf.invoke(default.clj:66)"
    "query_processor.context$runf.invokeStatic(context.clj:45)"
    "query_processor.context$runf.invoke(context.clj:39)"
    "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
    "query_processor.reducible$pivot.invoke(reducible.clj:31)"
    "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__49090.invoke(mbql_to_native.clj:25)"
    "query_processor.middleware.check_features$check_features$fn__49576.invoke(check_features.clj:39)"
    "query_processor.middleware.limit$limit$fn__47463.invoke(limit.clj:37)"
    "query_processor.middleware.cache$maybe_return_cached_results$fn__49950.invoke(cache.clj:204)"
    "query_processor.middleware.optimize_temporal_filters$optimize_temporal_filters$fn__50943.invoke(optimize_temporal_filters.clj:204)"
    "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing$fn__50987.invoke(validate_temporal_bucketing.clj:50)"
    "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__49147.invoke(auto_parse_filter_values.clj:43)"
    "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__39249.invoke(wrap_value_literals.clj:161)"
    "query_processor.middleware.annotate$add_column_info$fn__43680.invoke(annotate.clj:615)"
    "query_processor.middleware.permissions$check_query_permissions$fn__45618.invoke(permissions.clj:108)"
    "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__50127.invoke(pre_alias_aggregations.clj:40)"
    "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46016.invoke(cumulative_aggregations.clj:60)"
    "query_processor.middleware.visualization_settings$update_viz_settings$fn__45954.invoke(visualization_settings.clj:63)"
    "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47061.invoke(resolve_joined_fields.clj:102)"
    "query_processor.middleware.resolve_joins$resolve_joins$fn__50713.invoke(resolve_joins.clj:171)"
    "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__50249.invoke(add_implicit_joins.clj:190)"
    "query_processor.middleware.large_int_id$convert_id_to_string$fn__47080.invoke(large_int_id.clj:59)"
    "query_processor.middleware.format_rows$format_rows$fn__50301.invoke(format_rows.clj:74)"
    "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__46452.invoke(add_default_temporal_unit.clj:23)"
    "query_processor.middleware.desugar$desugar$fn__45927.invoke(desugar.clj:21)"
    "query_processor.middleware.binning$update_binning_strategy$fn__39149.invoke(binning.clj:229)"
    "query_processor.middleware.resolve_fields$resolve_fields$fn__45261.invoke(resolve_fields.clj:34)"
    "query_processor.middleware.add_dimension_projections$add_remapping$fn__49502.invoke(add_dimension_projections.clj:312)"
    "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__49799.invoke(add_implicit_clauses.clj:147)"
    "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__47448.invoke(upgrade_field_literals.clj:40)"
    "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__46829.invoke(add_source_metadata.clj:123)"
    "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__50049.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
    "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__48399.invoke(auto_bucket_datetimes.clj:147)"
    "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45242.invoke(resolve_source_table.clj:45)"
    "query_processor.middleware.parameters$substitute_parameters$fn__48053.invoke(parameters.clj:111)"
    "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__45315.invoke(resolve_referenced.clj:79)"
    "query_processor.middleware.expand_macros$expand_macros$fn__51371.invoke(expand_macros.clj:184)"
    "query_processor.middleware.add_timezone_info$add_timezone_info$fn__47832.invoke(add_timezone_info.clj:15)"
    "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__50316.invoke(splice_params_in_response.clj:32)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49810$fn__49814.invoke(resolve_database_and_driver.clj:31)"
    "driver$do_with_driver.invokeStatic(driver.clj:60)"
    "driver$do_with_driver.invoke(driver.clj:56)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49810.invoke(resolve_database_and_driver.clj:25)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__45865.invoke(fetch_source_query.clj:274)"
    "query_processor.middleware.store$initialize_store$fn__46053$fn__46054.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
    "query_processor.store$do_with_store.invoke(store.clj:38)"
    "query_processor.middleware.store$initialize_store$fn__46053.invoke(store.clj:10)"
    "query_processor.middleware.validate$validate_query$fn__50099.invoke(validate.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__50106.invoke(normalize_query.clj:22)"
    "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__47778.invoke(add_rows_truncated.clj:35)"
    "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49076.invoke(results_metadata.clj:147)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__47796.invoke(constraints.clj:42)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__50085.invoke(process_userland_query.clj:146)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__50396.invoke(catch_exceptions.clj:169)"
    "query_processor.reducible$async_qp$qp_STAR___42500$thunk__42501.invoke(reducible.clj:103)"
    "query_processor.reducible$async_qp$qp_STAR___42500$fn__42503.invoke(reducible.clj:108)"],
   :error_type :invalid-query,
   :ex-data
   {:sql
    "-- Metabase:: userID: 45 queryType: native queryHash: 1d16aee245ecfa955f0168e69169e4d9b55d73aab19116a9a1a5cb13d2ad4a51\nOMITTING QUERY",
    :params nil,
    :type :invalid-query}}],
 :state "70100",
 :error_type :invalid-query,
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :native,
  :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
  :native
  {:template-tags {},
   :query
   "OMITTING QUERY    "},
  :database 2,
  :parameters [],
  :async? true,
  :cache-ttl nil},
 :status :failed,
 :class java.sql.SQLTimeoutException,
 :stacktrace
 ["org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:47)"
  "org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:153)"
  "org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:274)"
  "org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:363)"
  "org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:501)"
  "com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)"
  "--> driver.sql_jdbc.execute$fn__52357.invokeStatic(execute.clj:368)"
  "driver.sql_jdbc.execute$fn__52357.invoke(execute.clj:366)"
  "driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invokeStatic(execute.clj:376)"
  "driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invoke(execute.clj:373)"
  "driver.sql_jdbc.execute$execute_reducible_query$fn__52437.invoke(execute.clj:502)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:501)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:487)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:496)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:487)"
  "driver.sql_jdbc$fn__85100.invokeStatic(sql_jdbc.clj:54)"
  "driver.sql_jdbc$fn__85100.invoke(sql_jdbc.clj:52)"
  "query_processor.context$executef.invokeStatic(context.clj:59)"
  "query_processor.context$executef.invoke(context.clj:48)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
  "query_processor.context.default$default_runf.invoke(default.clj:66)"
  "query_processor.context$runf.invokeStatic(context.clj:45)"
  "query_processor.context$runf.invoke(context.clj:39)"
  "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
  "query_processor.reducible$pivot.invoke(reducible.clj:31)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__49090.invoke(mbql_to_native.clj:25)"
  "query_processor.middleware.check_features$check_features$fn__49576.invoke(check_features.clj:39)"
  "query_processor.middleware.limit$limit$fn__47463.invoke(limit.clj:37)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__49950.invoke(cache.clj:204)"
  "query_processor.middleware.optimize_temporal_filters$optimize_temporal_filters$fn__50943.invoke(optimize_temporal_filters.clj:204)"
  "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing$fn__50987.invoke(validate_temporal_bucketing.clj:50)"
  "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__49147.invoke(auto_parse_filter_values.clj:43)"
  "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__39249.invoke(wrap_value_literals.clj:161)"
  "query_processor.middleware.annotate$add_column_info$fn__43680.invoke(annotate.clj:615)"
  "query_processor.middleware.permissions$check_query_permissions$fn__45618.invoke(permissions.clj:108)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__50127.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46016.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.visualization_settings$update_viz_settings$fn__45954.invoke(visualization_settings.clj:63)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47061.invoke(resolve_joined_fields.clj:102)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__50713.invoke(resolve_joins.clj:171)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__50249.invoke(add_implicit_joins.clj:190)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__47080.invoke(large_int_id.clj:59)"
  "query_processor.middleware.format_rows$format_rows$fn__50301.invoke(format_rows.clj:74)"
  "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__46452.invoke(add_default_temporal_unit.clj:23)"
  "query_processor.middleware.desugar$desugar$fn__45927.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__39149.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__45261.invoke(resolve_fields.clj:34)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__49502.invoke(add_dimension_projections.clj:312)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__49799.invoke(add_implicit_clauses.clj:147)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__47448.invoke(upgrade_field_literals.clj:40)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__46829.invoke(add_source_metadata.clj:123)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__50049.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__48399.invoke(auto_bucket_datetimes.clj:147)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45242.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__48053.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__45315.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__51371.invoke(expand_macros.clj:184)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__47832.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__50316.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49810$fn__49814.invoke(resolve_database_and_driver.clj:31)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49810.invoke(resolve_database_and_driver.clj:25)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__45865.invoke(fetch_source_query.clj:274)"
  "query_processor.middleware.store$initialize_store$fn__46053$fn__46054.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__46053.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__50099.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__50106.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__47778.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49076.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47796.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__50085.invoke(process_userland_query.clj:146)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__50396.invoke(catch_exceptions.clj:169)"
  "query_processor.reducible$async_qp$qp_STAR___42500$thunk__42501.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___42500$fn__42503.invoke(reducible.clj:108)"],
 :card_id 2836,
 :context :question,
 :error "(conn=747412) Query execution was interrupted",
 :row_count 0,
 :running_time 0,
 :data {:rows [], :cols []}}
```

**To Reproduce**
1. Spin up a metabase instance in elastic beanstalk
2. Add a MySQL 8 database
3. Create a query that takes longer than 60 seconds to execute
4. Run it

**Expected behavior**
It finishes successfully

**Screenshots**

Getting a generic "Your question took too long"

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, Safari 11.1, …)
- Your operating system: (e.x. OS X 10.10, Windows 10.1809, Ubuntu 16.04, …)
- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
- Metabase version: (e.x. 0.19.3)
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Jar-file on Ubuntu, …)
- Metabase internal database: (e.x. H2 (default), Postgres or MySQL)
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.13+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.13",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.13+8",
    "os.name": "Linux",
    "os.version": "4.14.252-131.483.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "snowflake",
      "mysql",
      "googleanalytics"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.12"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.23"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-12-10",
      "tag": "v0.41.4",
      "branch": "release-x.41.4",
      "hash": "471cfb9"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
This is pretty bad, since we run a lot of long-running queries that we now have to rewrite to break up into smaller queries until we figure out what's going on.

**Additional context**
Might be a red herring, but i saw https://github.com/metabase/metabase/blob/badb0bfaf275915ba8850406afc6f563a4d86b60/src/metabase/query_processor/context/default.clj#L16-L19 and wondered if maybe the docker image was configured w/ test vs prod(?)

Edit: I think this pr is related https://github.com/metabase/metabase/pull/17261/files it deletes nginx configs in the .ebextensions directory, which for AL 1 Docker is necessary (AL2 uses .platform).

Edit2: Im testing a fix..but if i read the PR right above, nginx isn't needed anymore for AL1 deployments? I'm gonna try launching without nginx to see if that fixes my problem, but if that's the case you should definitely update the elastic beanstalk tutorials for folks still on AL1 Docker.

Edit3: Trying to disable nginx as a reverse-proxy caused the metabase instances to not be able to connect to the metabase DB (which i setup as a 'coupled database' in EB). I've no clue why the reverse-proxy would do that, but clearly it's not a solution. Short of upgrading to AL2, IDK if there's a good solution/workaround for AL1. (Edit5: I realized my issue was that i wasn't including all the original files in the EB-MB zip; my guess is i could get this to work, but it's not worth the effort given what i ended up doing while keeping nginx).

Edit4: I finally was able to fix it! Once i realized that the EB zip contained not only the dockerrun json _but also the eb config dirs_, i just modified metabase-setup.sh and 01_metabase.config under .ebextensions to add back the server_https function. I might submit a PR (at least to illustrate the fix to others, so someone else can maybe clean it up + test it more properly)

**Comment by camsaul** [comment]
This timeout is coming from the MariaDB JDBC driver itself, based on this part of the error

```
:class java.sql.SQLTimeoutException,
 :stacktrace
 ["org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:47)"
```

I'm guessing it has a default timeout of 60 seconds. We should override this so it matches that 20 minute timeout we normally want to use in prod. Probably just a connection option or something

**Comment by bshepherdson** [comment]
Metabase treats MySQL and MariaDB nearly identically, but this is one place where they diverge.

[Gory details here](https://mariadb.com/kb/en/aborting-statements/#differences-between-the-mariadb-and-mysql-implementations)
- MySQL uses `max_execution_time` in milliseconds
- MariaDB uses `max_statement_time` in seconds

Further, `SET SESSION` queries to override session variables don't work in read-only sessions like Metabase creates.

Once #13444 is in place, this can be done more easily. (By using the known DB type after the initial sync to set the right parameters when connecting.)

**Comment by joshwolff1** [comment]
Hi, so this is totally random, but I've been using Beanstalk for years (not a Metabase user).

I've been trying to figure out this problem for 30-40+ hours where Beanstalk terminates the connection after 60 seconds, irrespective of nginx/no nginx, nginx settings, load balance idle timeout, etc. etc. 

Thought I'd post this note to possibly help you guys, in some way.

**Comment by paoliniluis** [comment]
We have deprecated shipping the Beanstalk artifacts, so this is no longer relevant. Users will have to build their own beanstalk artifacts from now on
